src/connectors/toggleRefinement/connectToggleRefinement.js

```diff
@@ -277,23 +274,38 @@ export default function connectToggleRefinement(renderFn, unmountFn = noop) {
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
+        const withFacetConfiguration = searchParameters.setQueryParameters({
+          disjunctiveFacets: [attribute],
+          disjunctiveFacetsRefinements: {
+            [attribute]: [],
+          },
+        });
+
         const isRefined = Boolean(uiState.toggle && uiState.toggle[attribute]);
 
         if (isRefined) {
-          if (hasAnOffValue)
-            return searchParameters
+          if (hasAnOffValue) {
```

Comment:
We don't have to branch to remove the value we can always call remove even when the value isn't there. The function can be simplified if we move the removal part above the branches. Right now it's a bit tedious to follow.

Reply:
a0642b475ec76a6e9fb0bcd7a77d44e2194fdef2

I didn't touch this logic though. Hopefully that's what you mean.